packages/tic-tac-toe/contracts/TicTacToe.sol

```diff
@@ -0,0 +1,617 @@
+        stakeUnchanged(fromGameData, toGameData)
+    {
+        require(toGameData.Os == 0, 'Os reset to 0');
+        require(toGameData.Xs == 0, 'Xs reset to 0');
+    }
+
+    function requireValidVICTORYtoSTART(
+        Outcome.AllocationItem[] memory fromAllocation,
+        Outcome.AllocationItem[] memory toAllocation,
+        TTTData memory fromGameData,
+        TTTData memory toGameData
+    )
+        private
+        pure
+        allocationUnchanged(fromAllocation, toAllocation)
+        stakeUnchanged(fromGameData, toGameData)
+    {
+        require(toGameData.Os == 0, 'Os reset to 0');
+        require(toGameData.Xs == 0, 'Xs reset to 0');
+
+    }
+
+    function extractAllocation(VariablePart memory variablePart)
+        private
+        pure
+        returns (Outcome.AllocationItem[] memory)
+    {
+        Outcome.OutcomeItem[] memory outcome = abi.decode(variablePart.outcome, (Outcome.OutcomeItem[]));
+        require(outcome.length == 1, 'TicTacToe: Only one asset allowed');
+
+        Outcome.AssetOutcome memory assetOutcome = abi.decode(
+            outcome[0].assetOutcomeBytes,
+            (Outcome.AssetOutcome)
+        );
+
+        require(
+            assetOutcome.assetOutcomeType == uint8(Outcome.AssetOutcomeType.Allocation),
+            'TicTacToe: AssetOutcomeType must be Allocation'
+        );
+
+        Outcome.AllocationItem[] memory allocation = abi.decode(
+            assetOutcome.allocationOrGuaranteeBytes,
+            (Outcome.AllocationItem[])
+        );
+
+        require(
+            allocation.length == 2,
+            'TicTacToe: Allocation length must equal number of participants (i.e. 2)'
+        );
+
+        return allocation;
+    }
+
+    function winnings(uint256 currentWinnerPlayer, uint256 stake)
+        private
+        pure
+        returns (uint256, uint256)
+    {
+        if (currentWinnerPlayer == 0) {
+            // first player won
+            return (2 * stake, 0);
+        } else if (currentWinnerPlayer == 1) {
+            // second player won
+            return (0, 2 * stake);
+        } else {
+            // Draw
+            return (stake, stake);
+        }
+    }
+
+    function _requireDestinationsUnchanged(
+        Outcome.AllocationItem[] memory fromAllocation,
+        Outcome.AllocationItem[] memory toAllocation
+    ) private pure {
+        require(
+            toAllocation[0].destination == fromAllocation[0].destination,
+            'TicTacToe: Destimation playerA may not change'
+        );
+        require(
+            toAllocation[1].destination == fromAllocation[1].destination,
+            'TicTacToe: Destimation playerB may not change'
+        );
+    }
+
+    // modifiers
+    modifier outcomeUnchanged(VariablePart memory a, VariablePart memory b) {
+        require(keccak256(b.outcome) == keccak256(a.outcome), 'TicTacToe: Outcome must not change');
+        _;
+    }
+
+    modifier noDisjointMoves(TTTData memory toGameData) {
```

Comment:
Do you think `disjointMarks` might be a better name for this modifier?  We don't want to ensure *no* disjoint moves.